### PR TITLE
ADC: Various lmp90xxx fixes and improvements

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
This series of patches fixes and improved the TI LMP90xxx ADC/GPIO driver. Summary of the changes:
- Retry readings on failure in the lmp90100_evb sample
- Keep the acquisition thread running even if an ADC reading failed (bug)
- Use common crc8 implementation
- Make adc_context_start_sampling() ISR safe
- Allow adc_read_async() to be called from ISR context
- Validate ADC sequence channels mask
- Prevent SPI transactions from ISR context
- Convert drivers to new DT API

Please look at each individual commit for further details. I bundled these in one PR but I can split out the DT API stuff if needed.

### CODE REVIEW

File: drivers/adc/adc_lmp90xxx.c [hunk 1]
```diff
@@ -1098,142 +1117,54 @@ static const struct adc_driver_api lmp90xxx_adc_api = {
  * LMP90077: 16 bit, 2 diff/4 se (4 channels), 0 currents
  */
 #if DT_INST_0_TI_LMP90077
```
Comment (galak) [inline drivers/adc/adc_lmp90xxx.c]: These #if should be something like `#if DT_HAS_COMPAT(ti_lmp90077)`

**Review by galak — CHANGES_REQUESTED** [review]
The ifdef's for the various compat's needs updating from #if DT_INST_0_<COMPAT> to #if DT_HAS_COMPAT(compat)

File: drivers/adc/adc_lmp90xxx.c (hunk 1 above)
Comment (henrikbrixandersen) [reply drivers/adc/adc_lmp90xxx.c]: Fixed, thank you.

**Review by galak — APPROVED** [review]
DT changes look good.

**Pull request merged**